### PR TITLE
feat: run a scheduler only once at startup

### PR DESCRIPTION
**Pull request by zhennann**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

### CODE REVIEW

**Comment by zhennann** [comment]
@atian25 Why "continuous-integration/appveyor/branch" always not passed, something wrong?

File: lib/timer.js [hunk 1]
```diff
@@ -19,8 +19,8 @@ class Timer {
    */
   handler(key, schedule, listener) {
     const { interval, cron, cronOptions, immediate } = schedule;
-    if (!interval && !cron) {
-      throw new Error('[egg-schedule] schedule.interval or schedule.cron must be present');
+    if (!interval && !cron && !immediate) {
```
Comment (atian25) [inline lib/timer.js]: I had another idea, we should change this check to `assert`.

```js
assert(interval || cron || immediate, '[egg-schedule] schedule.interval or schedule.cron or schedule.immediate must be present')
```

**Comment by atian25** [comment]
> @atian25 Why "continuous-integration/appveyor/branch" always not passed, something wrong?

maybe it's permission problem, forgot it.

File: lib/timer.js (hunk 1 above)
Comment (zhennann) [reply lib/timer.js]: ok, it's well.

**Comment by atian25** [comment]
3.1.0

**Pull request merged**